tests/unit/test_dr_client.py

```diff
@@ -162,6 +174,22 @@ def _generate_for_single_page(page_index, num_entities, has_next):
     return total_entities
 
 
+def mock_single_page_response(url, entities, match=None):
+    """Mock single page paginated response."""
+
+    def url_factory(_):
```

Comment:
I don't see how they are the same.. I can't use `paginated_url_factory` as an argument to `mock_paginated_response` since it expects a factory that takes a single page number parameter.

But after taking a look I found that instead of using the fixture I created: [fixture_url_factory](https://github.com/datarobot-oss/custom-models-action/pull/355/files/4c1708227955fd47a14ec180023fa72d536dff5c#diff-d2a57fd330cfb981e92f0aa7b3b4cd3f622a19bfd0d8ac48ed8706da8cec38a1R50), I can just use `paginated_url_factory`